aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp

```diff
@@ -563,12 +605,20 @@ class QEmbedding final {
       const c10::intrusive_ptr<EmbeddingPackedParamsBase>& packed_weight,
       const Tensor& indices,
       bool pruned_weights) {
+    // Set default offsets here since the FBGEMM lookup op expects it.
     const auto offsets_size = indices.numel();
-    at::Tensor offsets = at::arange(0, offsets_size, at::kLong);
+    at::Tensor offsets = at::arange(0, offsets_size, indices.scalar_type());
     at::Tensor output;
     if (bit_rate == 8) {
-      return packed_weight->embeddingbag_byte(
-          indices, offsets, pruned_weights, c10::nullopt, c10::nullopt, false);
+      return embedding_bag_byte_helper(
+          packed_weight->get_packed_weight().contiguous(),
```

Comment:
I think the common practice is to define embeddingbag_byte on packed_weight since packed_weight can be in different backends (fbgemm/qnnpack), examples can be found in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qconv.cpp. why do we change this?

Reply:
This can be changed. The main motivation was to not have to change the definition of `embeddingbag_byte` operator to add a new bool argument (but it is doable). 
In this case the packed weight is still stored as a PyTorch tensor so it can be directly accessed.

Reply:
I see, if you need to distinguish between embedding/emedding_bag in the invocation to packed weight, I think you can follow ConvPackedParams as well, we can add a new template argument, (True/False) maybe to distinguish between these two cases. there might be implications for serialization if we change this though.

Reply:
i think for now I can add a bool argument, since template might make it too complex for this PR.